### PR TITLE
[#9356] Improve UI for loading indicators - Loading bar

### PR DESCRIPTION
**Pull request by erjillsison**
Part of #9356 

The loading-bar is used in the following pages:
- Instructor home page
- Instructor student list page 
- Instructor search page
- Student home page

For the home page and student list page, the loading bar is used during the initial loading of the course cards. It's not used during the loading of the feedback sessions within a course card (that is covered by PR #9810).

When testing locally, the change is hardly noticeable. To test, I made use of `.pipe(delay(1234))` attached to the http-request observable.

loading home student list

**Additional change:**
- Hide the "New to teammates..." message in instructor home page while the page is still loading.

**Outline of solution**
The `loader-bar component` is placed in`page-component`.
To activate the loading bar, a `loading-bar-service` is used which has two methods: `showLoadingBar()` and `hideLoadingBar()`.

### CODE REVIEW

**Comment by erjillsison** [comment]
Ready for review

**Comment by crizyli** [comment]
LGTM!

File: src/web/app/pages-instructor/instructor-search-page/instructor-search-page.component.ts [hunk 1]
```diff
@@ -69,22 +73,25 @@ export class InstructorSearchPageComponent implements OnInit {
    * Searches for students and questions/responses/comments matching the search query.
    */
   search(searchQuery: SearchQuery): void {
+    this.loadingBarService.showLoadingBar();
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-search-page/instructor-search-page.component.ts]: So every time I make a request I shall call `this.loadingBarService.showLoadingBar();` and then `this.loadingBarService.hideLoadingBar()`?

Comment (erjillsison) [reply src/web/app/pages-instructor/instructor-search-page/instructor-search-page.component.ts]: Yes.

Comment (xpdavid) [reply src/web/app/pages-instructor/instructor-search-page/instructor-search-page.component.ts]: Why no inject it to httpservice?

Comment (erjillsison) [reply src/web/app/pages-instructor/instructor-search-page/instructor-search-page.component.ts]: I misunderstood the initial question. The above methods are only called on certain request calls and not on all.

File: src/web/app/pages-instructor/instructor-search-page/instructor-search-page.component.ts [hunk 2]
```diff
@@ -69,22 +73,25 @@ export class InstructorSearchPageComponent implements OnInit {
...
-      }
-    }, (resp: ErrorMessageOutput) => {
-      this.statusMessageService.showErrorMessage(resp.error.message);
-    });
+    this.httpRequestService.get('/studentsAndSessionData/search', paramMap)
+        .pipe(finalize(() => this.loadingBarService.hideLoadingBar()))
+        .subscribe((resp: SearchResult) => {
+          this.studentTables = resp.searchStudentsTables;
+          this.fbSessionDataTables = resp.searchFeedbackSessionDataTables;
+          const hasStudents: boolean = !!(this.studentTables && this.studentTables.length);
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-search-page/instructor-search-page.component.ts]: Is this like a double-negative, which is the same as `this.studentTables && this.studentTables.length`?

**Pull request merged**